### PR TITLE
output/reference: Include reference information in alert (if configured)

### PR DESCRIPTION
**Pull request by jlucovsky**
When configured, include the reference value in the alert. The configuration value is in the `alert` section:  types.alert.reference. The default value is off/no. Set to yes to include the expanded reference from the rule in the alert record.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [4974](https://redmine.openinfosecfoundation.org/issues/4974)

Describe changes:
- Add `reference` value to suricata.yaml.in (default no/off)
- Set flag in output logger if the config setting is on
- Format the reference as a sequence, e.g., `references: [ "ref-1" [, "ref-2" [, ...]]]`

### Provide values to any of the below to override the defaults.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1808

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .uptime | 639 | 661 | 103.44% | 

Pipeline 20407

File: src/output-json-alert.c [hunk 1]
```diff
@@ -173,6 +174,25 @@ static void AlertJsonSourceTarget(const Packet *p, const PacketAlert *pa,
...
+static void AlertJsonReference(
+        AlertJsonOutputCtx *json_output_ctx, const PacketAlert *pa, JsonBuilder *jb)
+{
+    if (!pa->s->references) {
+        return;
+    }
+
+    const DetectReference *kv = pa->s->references;
+    jb_open_array(jb, "references");
+    while (kv) {
```
Comment (catenacyber) [inline src/output-json-alert.c:186]: How are sure the array is not empty ?

File: src/output-json-alert.c [hunk 2]
```diff
@@ -173,6 +174,25 @@ static void AlertJsonSourceTarget(const Packet *p, const PacketAlert *pa,
...
+        const size_t size = strlen(kv->key) + strlen(kv->reference) + 1;
+        char kv_store[size];
+        snprintf(kv_store, size, "%s%s", kv->key, kv->reference);
```
Comment (catenacyber) [inline src/output-json-alert.c:189]: Should there not be a separator like `:` between key and value ?

Comment (jlucovsky) [reply src/output-json-alert.c:189]: No ... An example key is `https://` and the corresponding value might be `github.com/OISF/suricata`

File: src/output-json-alert.c (hunk 1 above)
Comment (jlucovsky) [reply src/output-json-alert.c:186]: Good point. I'll fix that.

File: src/output-json-alert.c (hunk 2 above)
Comment (catenacyber) [reply src/output-json-alert.c:189]: So, key is rather a protocol scheme ?

**Review by catenacyber — CHANGES_REQUESTED** [review]
Need to handle empty arrays as discussed

File: src/output-json-alert.c (hunk 2 above)
Comment (jasonish) [reply src/output-json-alert.c:189]: Isn't the scheme more like `url`, `md5`, `cve`?

**Pull request closed without merging**